lib/xgu/xgu.h

```diff
@@ -0,0 +1,937 @@
+                                  XGU_MASK(NV097_SET_TEXTURE_FORMAT_BASE_SIZE_U, u_size) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FORMAT_BASE_SIZE_V, v_size) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FORMAT_BASE_SIZE_P, p_size));
+}
+
+inline
+uint32_t* xgu_set_texture_address(uint32_t* p, uint32_t address) {
+    return push_command_parameter(p, NV097_SET_TEXTURE_ADDRESS, address);
+}
+
+inline
+uint32_t* xgu_set_texture_control0(uint32_t* p, bool enable, uint16_t min_lod, uint16_t max_lod) {
+    return push_command_parameter(p, NV097_SET_TEXTURE_CONTROL0,
+                                  (enable ? NV097_SET_TEXTURE_CONTROL0_ENABLE : 0) |
+                                  XGU_MASK(NV097_SET_TEXTURE_CONTROL0_MIN_LOD_CLAMP, min_lod) |
+                                  XGU_MASK(NV097_SET_TEXTURE_CONTROL0_MAX_LOD_CLAM, max_lod));
+}
+
+inline
+uint32_t* xgu_set_texture_control1(uint32_t* p, uint16_t pitch) {
+    return push_command_parameter(p, NV097_SET_TEXTURE_CONTROL1,
+                                  XGU_MASK(NV097_SET_TEXTURE_CONTROL1_IMAGE_PITCH, pitch));
+}
+
+inline
+uint32_t* xgu_set_texture_filter(uint32_t* p, uint16_t lod_bias, uint8_t filter_min, uint8_t filter_mag,
+                                 bool alpha_signed, bool r_signed, bool g_signed, bool b_signed) {
+    return push_command_parameter(p, NV097_SET_TEXTURE_FILTER,
+                                  XGU_MASK(NV097_SET_TEXTURE_FILTER_MIPMAP_LOD_BIAS, lod_bias) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FILTER_MIN, filter_min) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FILTER_MAG, filter_mag) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FILTER_ASIGNED, alpha_signed) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FILTER_RSIGNED, r_signed) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FILTER_GSIGNED, g_signed) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FILTER_BSIGNED, b_signed));
+}
+
+inline
+uint32_t* xgu_set_texture_image_rect(uint32_t* p, uint16_t width, uint16_t height) {
+    return push_command_parameter(p, NV097_SET_TEXTURE_IMAGE_RECT,
+                                  XGU_MASK(NV097_SET_TEXTURE_IMAGE_RECT_WIDTH, width) |
+                                  XGU_MASK(NV097_SET_TEXTURE_IMAGE_RECT_HEIGHT, height));
+}
+
+inline
+uint32_t* xgu_set_texture_palette(uint32_t* p, bool context_dma, XguPaletteLen palette_length,
+                                  uint32_t offset) {
+    return push_command_parameter(p, NV097_SET_TEXTURE_palette,
+                                  XGU_MASK(NV097_SET_TEXTURE_PALETTE_CONTEXT_DMA, context_dma) |
+                                  XGU_MASK(NV097_SET_TEXTURE_PALETTE_LENGTH, palette_length) |
+                                  XGU_MASK(NV097_SET_TEXTURE_PALETTE_OFFSET, offset));
+}
+
+inline
+uint32_t* xgu_set_texture_border_color(uint32_t* p, uint32_t color) {
```

Comment:
Should probably be specified as r,g,b,a - although it might be encoded in the same format as the texture pixels? Needs testing probably - I don't think XQEMU emulates this yet either.